product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteExistingAPIsTestCases.java

```diff
@@ -0,0 +1,233 @@
+    public void testDeleteAPI(APILifeCycleState state) throws Exception {
+        String name = API_NAME_PREFIX + state.toString();
+        createApi(name);
+        changeApiStateTo(name, state);
+        deleteAPI(name);
+    }
+
+    @Test(description = "1.4.1.2", dataProvider = "DeleteAPIAfterSubscribingDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testDeleteAPIAfterDeleteSubscribedApplication(APILifeCycleState state) throws Exception {
+        String name = API_NAME_PREFIX + "deleteSubApp_" + state.toString();
+        createAPIAndSubscribe(name, name);
+        if (state.equals(APILifeCycleState.BLOCKED) || state.equals(APILifeCycleState.DEPRECATED)) {
+            changeApiState(name, state);
+        }
+        verifyResponse(apiStore.removeApplication(name));
+        deleteAPI(name);
+    }
+
+    @Test(description = "1.4.1.3", dataProvider = "DeleteAPIAfterSubscribingDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testDeleteAPIAfterUnsubscribeApplication(APILifeCycleState state) throws Exception {
+        String name = API_NAME_PREFIX + "unsubApp_" + state.toString();
+        createAPIAndSubscribe(name, name);
+        if (state.equals(APILifeCycleState.BLOCKED) || state.equals(APILifeCycleState.DEPRECATED)) {
+            changeApiState(name, state);
+        }
+        verifyResponse(apiStore.removeAPISubscription(name, API_VERSION, API_CREATOR_PUBLISHER_USERNAME,
+                apiStore.getApplicationId(name)));
+        deleteAPI(name);
+    }
+
+    private void createApi(String apiName) throws Exception {
+        APIRequest apiRequest = new APIRequest(apiName, "/" + apiName, "public", API_VERSION,
+                "/menu", APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED,
+                new URL("https://localhost:9443/am/sample/pizzashack/v1/api/"));
+        HttpResponse createAPIResponse = apiPublisher.addAPI(apiRequest);
+        apiList.add(apiName);
+        verifyResponse(createAPIResponse);
+        HttpResponse apiInfo = apiPublisher.getAPI(apiName, API_CREATOR_PUBLISHER_USERNAME, API_VERSION);
+        verifyResponse(apiInfo);
+    }
+
+    private void changeApiStateTo(String apiName, APILifeCycleState state) throws Exception{
+        switch (state) {
+            case PROTOTYPED:
+            case PUBLISHED:
+                changeApiState(apiName, state);
+                break;
+            case BLOCKED:
+            case DEPRECATED:
+                changeApiState(apiName, APILifeCycleState.PUBLISHED);
+                changeApiState(apiName, state);
+                break;
+            case RETIRED:
+                changeApiState(apiName, APILifeCycleState.PUBLISHED);
+                changeApiState(apiName, APILifeCycleState.DEPRECATED);
+                changeApiState(apiName, state);
+                break;
+            case CREATED:
+            default:
```

Comment:
There is no use of defining  'default' case, if you do not do anything in it.

Reply:
Fixed with #5210